api/service.go

```diff
@@ -66,11 +68,15 @@ func (s *Service) Close() error {
 func (s *Service) NewDB(ctx context.Context, req *pb.NewDBRequest) (*pb.NewDBReply, error) {
 	log.Debugf("received new db request")
 
-	creds, err := getCredentials(req.Credentials)
+	id, err := thread.Cast(req.Body.DbID)
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, err.Error())
+	}
+	auth, err := getAuth(req.Header, req.Body)
```

Comment:
Thought I read your comments about moving to metadata, but I guess not yet?

Reply:
You can see the approach in the `net/api`. I wanted to get some confirmation on my signing approach before making changes to `api`. Plus, I ran out of time before bed.

Reply:
Cool, I like what I saw in net